src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -34,7 +47,13 @@
 public class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    try {
+      List<YoutubePost> newPost = YoutubeApi.getYoutubePost();
+      response.setContentType("text/html;");
+      response.getWriter().println(newPost);
+    } catch (IOException | YoutubeApiException e) {
+      System.out.println("Error: Youtube api returning exception" + e);
+      response.sendError(500);
```

Comment:
Could you include a message with the error to provide a bit more context on the 500? it's the second argument of `sendError`